src/Interpreters/MergeJoin.cpp

```diff
@@ -778,7 +779,10 @@ void MergeJoin::joinSortedBlock(Block & block, ExtraBlockPtr & not_processed)
                 if (intersection < 0)
                     break; /// (left) ... (right)
                 if (intersection > 0)
+                {
+                    skip_right = 0;
                     continue; /// (right) ... (left)
```

Comment:
If not reset skip_right, may be cause wrong right cursor position in the following `leftJoin<is_all>(...)`

Reply:
@vdimir